test/Dialect/FIRRTL/annotations.fir

```diff
@@ -852,3 +855,18 @@ circuit Foo : %[[
     ; CHECK-SAME: "sifive.enterprise.grandcentral.AugmentedBooleanType"
     ; CHECK-SAME: "sifive.enterprise.grandcentral.AugmentedIntegerType"
     ; CHECK-SAME: "sifive.enterprise.grandcentral.AugmentedDoubleType"
+
+; // -----
+
+; A non-local Annotation is supported.
+
+circuit FooNLA: %[[{"a":"a","target":"~FooNLA|FooNLA/bar:Bar/baz:Baz"}]]
+  module Baz :
+    skip
+  module Bar :
+    inst baz of Baz
+  module FooNLA :
+    inst bar of Bar
+; CHECK-LABEL: firrtl.circuit "FooNLA"
+; CHECK: firrtl.module @Baz
+; CHECK-SAME: annotations = [{a = "a", circt.nonlocal = @"~FooNLA|FooNLA/bar:Bar/baz:Baz"}]
```

Comment:
This test should also be checking that the intermediary annotations show up along the instance path.

It would also be good to test that `~FooNLA|Bar/baz:Baz` works. (It does work already, I just want to lock it in. 😄 )